### PR TITLE
nixos/keycloak: fix missing quotes for admin password 

### PR DESCRIPTION
**Pull request by red-hood**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by mweinelt — APPROVED** [review]

**Review by RaitoBezarius — APPROVED** [review]
Thanks for the catch!

**Comment by mweinelt** [comment]
@ofborg test keycloak

File: nixos/modules/services/web-apps/keycloak.nix [hunk 1]
```diff
@@ -637,7 +637,7 @@ in
               cp $CREDENTIALS_DIRECTORY/ssl_{cert,key} /run/keycloak/ssl/
             '' + ''
               export KEYCLOAK_ADMIN=admin
-              export KEYCLOAK_ADMIN_PASSWORD=${cfg.initialAdminPassword}
+              export KEYCLOAK_ADMIN_PASSWORD="${cfg.initialAdminPassword}"
```
Comment (alyssais) [inline nixos/modules/services/web-apps/keycloak.nix:640]: I think this should maybe be

```suggestion
              export KEYCLOAK_ADMIN_PASSWORD=${lib.escapeShellArg cfg.initialAdminPassword}
```

Which should handle e.g. a password containing a `"`.

**Comment by red-hood** [comment]
Sorry for the delay, but it took some time to dig deeper and understand the layers of nix, python and shell used in the tests. I created the changes necessary to support arbitrary combination of quotes on a separate branch: https://github.com/NixOS/nixpkgs/compare/master...red-hood:nixpkgs:feat_keycloak_pwquote
I personally think that the combination of string templating in layers of three different languages makes these changes quite hard to read and to maintain. I would therefore opt for just quoting the variable in single quotes and add a statement in the documentation not to use any single quotes in the password.

**Comment by talyz** [comment]
I would absolutely prefer to go with the solution suggested by @alyssais. Here is my complete solution to the python double escaping problem: https://github.com/talyz/nixpkgs/commit/afa8533be32517b0d476f2ebea58af9e7bf79b57, i.e. read from a file instead of splicing in the actual password.

**Pull request closed without merging**